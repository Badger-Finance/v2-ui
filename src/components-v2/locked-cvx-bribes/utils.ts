import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
