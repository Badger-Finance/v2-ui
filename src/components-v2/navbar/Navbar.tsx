import React from 'react';
import { LayoutContainer } from '../common/Containers';
import { NavbarInfoRow } from './NavbarInfoRow';
import { Divider, makeStyles, useMediaQuery, useTheme } from '@material-ui/core';
import { NavbarActionsRow } from './NavbarActionsRow';
import { NavbarMobileRow } from './NavbarMobileRow';
import { NavbarStats } from './NavbarStats';
import { NavbarTabs } from './NavbarTabs';
import clsx from 'clsx';

const useStyles = makeStyles((theme) => ({
	root: {
		width: '100%',
		backgroundColor: '#2a2a2a',
		position: 'sticky',
		top: 0,
		zIndex: 100,
		[theme.breakpoints.up('md')]: {
			paddingTop: '15px',
		},
	},
	actions: {
		marginTop: 21,
	},
	mobileRow: {
		padding: '16px 40px',
		[theme.breakpoints.down('sm')]: {
			padding: '16px 30px',
		},
	},
	mobileStats: {
		position: 'relative',
		overflowX: 'auto',
		[theme.breakpoints.down('sm')]: {
			padding: '16px 10px 16px 30x',
		},
	},
	mobileTabs: {
		[theme.breakpoints.down('xs')]: {
			paddingTop: 30,
		},
	},
}));

export const Navbar = (): JSX.Element => {
	const classes = useStyles();
	const isMobile = useMediaQuery(useTheme().breakpoints.down('sm'));

	if (isMobile) {
		return (
			<div className={classes.root}>
				<div className={classes.mobileRow}>
					<NavbarMobileRow />
				</div>
				<Divider />
				<div className={clsx(classes.mobileRow, classes.mobileStats)}>
					<NavbarStats />
				</div>
				<Divider />
				<div className={classes.mobileTabs}>
					<NavbarTabs />
				</div>
			</div>
		);
	}

	return (
		<div className={classes.root}>
			<LayoutContainer>
				<NavbarInfoRow />
				<div className={classes.actions}>
					<NavbarActionsRow />
				</div>
			</LayoutContainer>
		</div>
	);
};
