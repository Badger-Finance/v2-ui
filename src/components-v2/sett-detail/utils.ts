import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
