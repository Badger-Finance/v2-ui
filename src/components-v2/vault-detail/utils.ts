import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
