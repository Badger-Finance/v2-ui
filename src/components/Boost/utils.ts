import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
