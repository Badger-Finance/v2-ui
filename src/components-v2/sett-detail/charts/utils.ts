import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
