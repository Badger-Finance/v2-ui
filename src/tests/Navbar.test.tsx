import { Chain } from 'mobx/model/network/chain';
import React from 'react';

import Navbar from '../components-v2/navbar';
import store from '../mobx/stores/RootStore';
import { createMatchMedia } from './Utils';
import { checkSnapshot } from './utils/snapshots';

/* eslint-disable */

class ResizeObserver {
  disconnect() {}
  observe() {}
  unobserve() {}
}

describe('Navbar', () => {
  beforeEach(() => {
    global.ResizeObserver = ResizeObserver;
    Chain.getChain(store.chain.network).deploy.token = '0x3472A5A71965499acd81997a54BBA8D852C6E53d';
    store.wallet.address = '0x1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a1a';
    store.prices.getPrice = jest.fn().mockReturnValue(80);
    store.tree.cycle = 5894;
    store.tree.lastUpdate = '0h 55m';
    store.vaults.protocolSummaryCache = {
      [store.chain.network]: {
        totalValue: 1_000_000,
        setts: [],
      },
    };
    store.router = {
      ...store.router,
      currentRoute: {
        path: '/',
        originalPath: '/',
        rootPath: '/',
        replaceUrlParams: jest.fn(),
        getRootPath: jest.fn(),
        getParamsObject: jest.fn(),
        component: React.createElement('div'),
      },
      goTo: jest.fn(),
      currentPath: '/',
    };
  });

  test('Renders correctly', async () => {
    checkSnapshot(<Navbar />);
  });

  test('Renders tablet version correctly', () => {
    window.matchMedia = createMatchMedia(900);
    checkSnapshot(<Navbar />);
  });

  test('Renders mobile version correctly', () => {
    window.matchMedia = createMatchMedia(480);
    checkSnapshot(<Navbar />);
  });
});
