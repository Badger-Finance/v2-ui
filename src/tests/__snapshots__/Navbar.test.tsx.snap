// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Renders correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiContainer-root-7 WithStyles(ForwardRef(Container))-root-6 MuiContainer-maxWidthLg-13"
    >
      <div
        class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-align-items-xs-center-26 MuiGrid-justify-content-xs-space-between-37"
      >
        <div
          class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-item-19 MuiGrid-align-items-xs-center-26 MuiGrid-grid-xs-9-60 MuiGrid-grid-md-8-87 MuiGrid-grid-lg-9-102"
        >
          <div
            class="MuiGrid-root-17 makeStyles-badgerIconContainer-15 MuiGrid-item-19"
          >
            <img
              alt="Badger Logo"
              class="makeStyles-badgerIcon-16"
              src="/assets/icons/badger_head.svg"
            />
          </div>
          <div
            class="MuiGrid-root-17 MuiGrid-item-19 MuiGrid-grid-xs-true-51"
          >
            <div
              class="MuiGrid-root-17 makeStyles-root-120 MuiGrid-container-18"
              id="here-ref"
            >
              <div
                class="MuiGrid-root-17 MuiGrid-item-19"
              >
                <span
                  class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
                >
                  BADGER Price:  
                </span>
                <div
                  class="MuiBox-root-157 MuiBox-root-158"
                >
                  <span
                    class="MuiTypography-root-125 makeStyles-helperText-124"
                  >
                    $80.00
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root-17 MuiGrid-item-19"
              >
                <span
                  class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
                >
                  Cycle: 
                  5894
                    
                  (last: 0h 55m)
                </span>
              </div>
              <div
                class="MuiGrid-root-17 MuiGrid-item-19"
              >
                <span
                  class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
                >
                  Ethereum
                   TVL:  
                </span>
                <div
                  class="MuiBox-root-157 MuiBox-root-161"
                >
                  <span
                    class="MuiTypography-root-125 makeStyles-helperText-124"
                  >
                    $1,000,000
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root-17 makeStyles-assets-122 MuiGrid-item-19"
              >
                <span
                  class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
                >
                  My Assets:  
                </span>
                <div
                  class="MuiBox-root-157 MuiBox-root-164"
                >
                  <span
                    class="MuiTypography-root-125 makeStyles-helperText-124"
                  >
                    $0
                  </span>
                </div>
              </div>
              <div
                class="MuiGrid-root-17 makeStyles-assets-122 MuiGrid-item-19"
              >
                <span
                  class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
                >
                  My APR:  
                </span>
                <div
                  class="MuiBox-root-157 MuiBox-root-167"
                >
                  <span
                    class="MuiTypography-root-125 makeStyles-helperText-124"
                  >
                    0%
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-item-19 MuiGrid-justify-content-xs-flex-end-36 MuiGrid-grid-xs-true-51"
        >
          <div
            class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-align-items-xs-center-26 MuiGrid-justify-content-xs-flex-end-36"
          >
            <div
              class="MuiGrid-root-17 makeStyles-iconContainer-169 MuiGrid-item-19"
            >
              <img
                alt="Discord Icon"
                class="makeStyles-socialIcon-168"
                src="/assets/icons/discord.svg"
              />
            </div>
            <div
              class="MuiGrid-root-17 makeStyles-iconContainer-169 makeStyles-twitter-170 MuiGrid-item-19"
            >
              <img
                alt="Twitter Icon"
                class="makeStyles-socialIcon-168"
                src="/assets/icons/twitter.svg"
              />
            </div>
            <div
              class="MuiGrid-root-17 MuiGrid-item-19"
            >
              <span
                class="MuiTypography-root-125 makeStyles-helperText-124"
              >
                <a
                  class="MuiTypography-root-125 MuiLink-root-172 MuiLink-underlineHover-174 MuiTypography-colorInherit-147"
                  href="https://docs.badger.com/"
                  rel="noopener"
                  target="_blank"
                >
                  DOCS
                </a>
              </span>
            </div>
            <div
              class="MuiGrid-root-17 makeStyles-governance-171 MuiGrid-item-19"
            >
              <span
                class="MuiTypography-root-125 makeStyles-helperText-124"
              >
                <a
                  class="MuiTypography-root-125 MuiLink-root-172 MuiLink-underlineHover-174 MuiTypography-colorInherit-147"
                  href="https://snapshot.org/#/badgerdao.eth"
                  rel="noopener"
                  target="_blank"
                >
                  GOVERNANCE
                </a>
              </span>
            </div>
            <div
              class="MuiGrid-root-17 MuiGrid-item-19"
            >
              <span
                class="MuiTypography-root-125 makeStyles-helperText-124"
              >
                <a
                  class="MuiTypography-root-125 MuiLink-root-172 MuiLink-underlineHover-174 MuiTypography-colorInherit-147"
                  href="https://forum.badger.finance"
                  rel="noopener"
                  target="_blank"
                >
                  FORUM
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-actions-2"
      >
        <div
          class="MuiGrid-root-17 MuiGrid-container-18"
        >
          <div
            class="MuiGrid-root-17 MuiGrid-item-19 MuiGrid-grid-xs-auto-50"
          >
            <div
              class="MuiTabs-root-180"
            >
              <div
                class="MuiTabs-scroller-185 MuiTabs-fixed-186"
                style="overflow: hidden;"
              >
                <div
                  class="MuiTabs-flexContainer-182"
                  role="tablist"
                >
                  <button
                    aria-selected="true"
                    class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194 MuiTab-selected-196"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiTab-wrapper-200"
                    >
                      VAULTS
                    </span>
                    <span
                      class="MuiTouchRipple-root-267"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiTab-wrapper-200"
                    >
                      BOOST
                    </span>
                    <span
                      class="MuiTouchRipple-root-267"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiTab-wrapper-200"
                    >
                      IBBTC
                    </span>
                    <span
                      class="MuiTouchRipple-root-267"
                    />
                  </button>
                </div>
                <span
                  class="PrivateTabIndicator-root-204 PrivateTabIndicator-colorPrimary-205 MuiTabs-indicator-190 makeStyles-indicator-179"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-item-19 MuiGrid-grid-xs-true-51"
          >
            <div
              class="MuiGrid-root-17 makeStyles-root-208 MuiGrid-container-18 MuiGrid-justify-content-xs-flex-end-36"
            >
              <div
                class="MuiGrid-root-17 MuiGrid-item-19"
              >
                <button
                  class="MuiButtonBase-root-201 MuiButton-root-212 MuiButton-outlined-217 makeStyles-button-209 makeStyles-loadingRewardsButton-211 MuiButton-outlinedPrimary-218"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label-213"
                  >
                    <div
                      class="makeStyles-loaderContainer-241 makeStyles-customSizing-243"
                    >
                      <div
                        class="MuiCircularProgress-root-244 MuiCircularProgress-colorPrimary-248 MuiCircularProgress-indeterminate-246"
                        role="progressbar"
                        style="width: 15px; height: 15px;"
                      >
                        <svg
                          class="MuiCircularProgress-svg-250"
                          viewBox="22 22 44 44"
                        >
                          <circle
                            class="MuiCircularProgress-circle-251 MuiCircularProgress-circleIndeterminate-253"
                            cx="44"
                            cy="44"
                            fill="none"
                            r="20.2"
                            stroke-width="3.6"
                          />
                        </svg>
                      </div>
                    </div>
                  </span>
                  <span
                    class="MuiTouchRipple-root-267"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root-17 MuiGrid-item-19"
              >
                <div>
                  <button
                    aria-label="open network selector"
                    class="MuiButtonBase-root-201 MuiButton-root-212 MuiButton-outlined-217 makeStyles-networkButton-258 MuiButton-outlinedPrimary-218"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label-213"
                    >
                      <img
                        alt="selected network icon"
                        class="makeStyles-selectedNetworkIcon-259"
                        src="/assets/icons/ethereum-network.svg"
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root-267"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiGrid-root-17 MuiGrid-item-19"
              >
                <button
                  class="MuiButtonBase-root-201 MuiButton-root-212 MuiButton-outlined-217 MuiButton-outlinedPrimary-218 MuiButton-disableElevation-223"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label-213 makeStyles-walletButtonLabel-264"
                  >
                    0x1..a1a
                  </span>
                  <span
                    class="MuiTouchRipple-root-267"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Navbar Renders mobile version correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-mobileRow-3"
    >
      <div
        class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-justify-content-xs-space-between-37"
      >
        <div
          class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-item-19 MuiGrid-align-items-xs-center-26 MuiGrid-grid-xs-true-51"
        >
          <button
            class="MuiButtonBase-root-201 MuiIconButton-root-270 makeStyles-menuIcon-267"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-278"
            >
              <img
                alt="open menu"
                src="/assets/icons/mobile-drawer-icon.svg"
              />
            </span>
            <span
              class="MuiTouchRipple-root-298"
            />
          </button>
          <img
            alt="Badger Logo"
            class="makeStyles-badgerIcon-268"
            src="/assets/icons/badger_head.svg"
          />
        </div>
        <div
          class="MuiGrid-root-17 makeStyles-buttons-269 MuiGrid-container-18 MuiGrid-spacing-xs-2-41 MuiGrid-item-19 MuiGrid-grid-xs-8-59"
        >
          <div
            class="MuiGrid-root-17 MuiGrid-item-19"
          >
            <div>
              <button
                aria-label="open network selector"
                class="MuiButtonBase-root-201 MuiButton-root-212 MuiButton-outlined-217 makeStyles-networkButton-258 MuiButton-outlinedPrimary-218"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-213"
                >
                  <img
                    alt="selected network icon"
                    class="makeStyles-selectedNetworkIcon-259"
                    src="/assets/icons/ethereum-network.svg"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root-298"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root-17 MuiGrid-item-19"
          >
            <button
              class="MuiButtonBase-root-201 MuiButton-root-212 MuiButton-outlined-217 MuiButton-outlinedPrimary-218 MuiButton-disableElevation-223"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-213 makeStyles-walletButtonLabel-264"
              >
                0x1..a1a
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root-279"
    />
    <div
      class="makeStyles-mobileRow-3 makeStyles-mobileStats-4"
    >
      <div
        class="MuiGrid-root-17 makeStyles-root-120 MuiGrid-container-18"
        id="here-ref"
      >
        <div
          class="MuiGrid-root-17 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            BADGER Price:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-288"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              $80.00
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root-17 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            Cycle: 
            5894
              
            (last: 0h 55m)
          </span>
        </div>
        <div
          class="MuiGrid-root-17 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            Ethereum
             TVL:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-291"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              $1,000,000
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root-17 makeStyles-assets-122 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            My Assets:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-294"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              $0
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root-17 makeStyles-assets-122 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            My APR:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-297"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              0%
            </span>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root-279"
    />
    <div
      class="makeStyles-mobileTabs-5"
    >
      <div
        class="MuiTabs-root-180"
      >
        <div
          class="MuiTabs-scroller-185 MuiTabs-fixed-186"
          style="overflow: hidden;"
        >
          <div
            class="MuiTabs-flexContainer-182"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194 MuiTab-selected-196 MuiTab-fullWidth-198"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper-200"
              >
                VAULTS
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194 MuiTab-fullWidth-198"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper-200"
              >
                BOOST
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194 MuiTab-fullWidth-198"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper-200"
              >
                IBBTC
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-204 PrivateTabIndicator-colorPrimary-205 MuiTabs-indicator-190 makeStyles-indicator-179"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Navbar Renders tablet version correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-mobileRow-3"
    >
      <div
        class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-justify-content-xs-space-between-37"
      >
        <div
          class="MuiGrid-root-17 MuiGrid-container-18 MuiGrid-item-19 MuiGrid-align-items-xs-center-26 MuiGrid-grid-xs-true-51"
        >
          <button
            class="MuiButtonBase-root-201 MuiIconButton-root-270 makeStyles-menuIcon-267"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label-278"
            >
              <img
                alt="open menu"
                src="/assets/icons/mobile-drawer-icon.svg"
              />
            </span>
            <span
              class="MuiTouchRipple-root-298"
            />
          </button>
          <img
            alt="Badger Logo"
            class="makeStyles-badgerIcon-268"
            src="/assets/icons/badger_head.svg"
          />
        </div>
        <div
          class="MuiGrid-root-17 makeStyles-buttons-269 MuiGrid-container-18 MuiGrid-spacing-xs-2-41 MuiGrid-item-19 MuiGrid-grid-xs-8-59"
        >
          <div
            class="MuiGrid-root-17 MuiGrid-item-19"
          >
            <div>
              <button
                aria-label="open network selector"
                class="MuiButtonBase-root-201 MuiButton-root-212 MuiButton-outlined-217 makeStyles-networkButton-258 MuiButton-outlinedPrimary-218"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-213"
                >
                  <img
                    alt="selected network icon"
                    class="makeStyles-selectedNetworkIcon-259"
                    src="/assets/icons/ethereum-network.svg"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root-298"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root-17 MuiGrid-item-19"
          >
            <button
              class="MuiButtonBase-root-201 MuiButton-root-212 MuiButton-outlined-217 MuiButton-outlinedPrimary-218 MuiButton-disableElevation-223"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label-213 makeStyles-walletButtonLabel-264"
              >
                0x1..a1a
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root-279"
    />
    <div
      class="makeStyles-mobileRow-3 makeStyles-mobileStats-4"
    >
      <div
        class="MuiGrid-root-17 makeStyles-root-120 MuiGrid-container-18"
        id="here-ref"
      >
        <div
          class="MuiGrid-root-17 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            BADGER Price:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-288"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              $80.00
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root-17 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            Cycle: 
            5894
              
            (last: 0h 55m)
          </span>
        </div>
        <div
          class="MuiGrid-root-17 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            Ethereum
             TVL:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-291"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              $1,000,000
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root-17 makeStyles-assets-122 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            My Assets:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-294"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              $0
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root-17 makeStyles-assets-122 MuiGrid-item-19"
        >
          <span
            class="MuiTypography-root-125 makeStyles-helperText-124 MuiTypography-displayInline-153"
          >
            My APR:  
          </span>
          <div
            class="MuiBox-root-157 MuiBox-root-297"
          >
            <span
              class="MuiTypography-root-125 makeStyles-helperText-124"
            >
              0%
            </span>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root-279"
    />
    <div
      class="makeStyles-mobileTabs-5"
    >
      <div
        class="MuiTabs-root-180"
      >
        <div
          class="MuiTabs-scroller-185 MuiTabs-fixed-186"
          style="overflow: hidden;"
        >
          <div
            class="MuiTabs-flexContainer-182"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194 MuiTab-selected-196"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper-200"
              >
                VAULTS
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper-200"
              >
                BOOST
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root-201 MuiTab-root-191 makeStyles-tab-178 MuiTab-textColorPrimary-194"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper-200"
              >
                IBBTC
              </span>
              <span
                class="MuiTouchRipple-root-298"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-204 PrivateTabIndicator-colorPrimary-205 MuiTabs-indicator-190 makeStyles-indicator-179"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
